graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py

```diff
@@ -2117,10 +2142,12 @@ def process_custom_mask_df(input_df: DataFrame, split_file: CustomSplit, mask_ty
                     col(split_file.mask_columns[0]).alias(f"custom_{mask_type}_mask_src"),
                     col(split_file.mask_columns[1]).alias(f"custom_{mask_type}_mask_dst"),
                 )
+                input_df_id = create_mapping(input_df)
                 join_condition = (
-                    input_df["src_str_id"] == custom_mask_df[f"custom_{mask_type}_mask_src"]
-                ) & (input_df["dst_str_id"] == custom_mask_df[f"custom_{mask_type}_mask_dst"])
-                mask_df = input_df.join(custom_mask_df, join_condition, "left_outer")
+                    input_df_id["src_str_id"] == custom_mask_df[f"custom_{mask_type}_mask_src"]
+                ) & (input_df_id["dst_str_id"] == custom_mask_df[f"custom_{mask_type}_mask_dst"])
+                mask_df = input_df_id.join(custom_mask_df, join_condition, "left_outer")
```

Comment:
The join here happens after the graph structure has been written to disk right? Then the order between the structure and masks will not match.

Reply:
Not yet. Because the next line we will do sorting to maintain the order.

Reply:
Edge structure is written to disk here: https://github.com/awslabs/graphstorm/blob/2fd1511a239f3530cdbde3ad3d4ec7b202e63a50/graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py#L1535-L1537 and labels are processed later here: https://github.com/awslabs/graphstorm/blob/2fd1511a239f3530cdbde3ad3d4ec7b202e63a50/graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py#L1575-L1581

I guess that we add the additional ID column before the join, within the mask processing, and re-order according to that, and that will guarantee the original order (same as was used when writing the edge structure) is maintained? Is my understanding correct?

Reply:
Yes. Correct.